src/cancellation_manager.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+
+#include "cancellation_manager.h"
+#include "utils.h"
+
+CancellationManager::CancellationManager(const std::string &cancellationFileName, const std::chrono::steady_clock::time_point &finishTime) :
+ finishTime(finishTime), cancellationFileName(cancellationFileName), checksToPerform(defaultChecksToPerformValue), performedChecks(0) {
+
+}
+
+bool CancellationManager::shouldStopExecution() const {
+    if (fileExists(cancellationFileName) || finishTime < std::chrono::steady_clock::now()) {
+        return true;
+    }
+    return false;
+}
+
+bool CancellationManager::shouldStopExecutionSyscallSafe() const {
```

Comment:
It might be unclear what method to use. Let's describe the difference in a comment.